# Fusarium venenatum Genome Assembly

======================================

This document details the commands used to assemble reference genome of fusarium venenatum

The following includes new approaches in the assembly of Fvenenatum genomes.

```
Note 1 - all this work was performed in the directory: /projects/fusarium_venenatum
Note 2 - Probably, an updated version of the tools used in this script can be found on the bioinformatics_tools repository.
```

## Data management

Data from two sequencing runs were merged together

```bash
# FYI - RAW READS
#/projects/oldhome/groups/harrisonlab/project_files/fusarium_venenatum/raw_dna/minion/F.venenatum/WT/F.venenatum_WT_07-03-17_albacore_v2.02.fastq.gz
#/projects/oldhome/groups/harrisonlab/project_files/fusarium_venenatum/raw_dna/minion/F.venenatum/WT/F.venenatum_WT_18-07-17_albacore_v2.02.fastq.gz

mkdir -p qc_dna/minion/F.venenatum/WT

cat F.venenatum_WT_07-03-17_albacore_v2.02_trim.fastq.gz F.venenatum_WT_18-07-17_albacore_v2.02_trim.fastq.gz > /projects/fusarium_venenatum/qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz
```


## Previous assemblies qc

This section contain qc from previous assemblies for comparison only.

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/previous_versions/F.venenatum/*/*.fa); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

```bash
  for Assembly in $(ls assembly/previous_versions/F.venenatum/WT_minion/WT_albacore_v2_contigs_softmasked_repeatmasker_TPSI_appended.fa); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

```bash
  for Assembly in $(ls assembly/previous_versions/F.venenatum/WT/WT_contigs_softmasked_repeatmasker_TPSI_appended.fa); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

```bash
for Assembly in $(ls assembly/previous_versions/F.venenatum/WT_minion/WT_albacore_v2_contigs_softmasked_repeatmasker_TPSI_appended.fa); do
Strain=WT_minion
Organism=F.venenatum
echo "$Organism - $Strain"
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Gene_prediction
BuscoDB=$(ls -d /projects/dbBusco/sordariomycetes_odb10)
OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd10
sbatch $ProgDir/busco.sh $Assembly $BuscoDB $OutDir
done
```

## Miniasm

```bash
# conda packages required: miniasm, minimap2, bbmap, busco
conda activate olc_assemblers
```

##### Assembly of uncorrected reads. Racon required after.

```bash
cd qc_dna/minion/F.venenatum/WT
# If reads have same names or same part splitted by space, fix them using rename.sh from bbtools.
rename.sh in=WT_minion_allfiles.fastq.gz out=WT_minion_allfiles.fasta prefix=WT_minion

# Fast all-against-all overlap of raw reads
# Overlap for MinION reads (or use "-x ava-pb" for Pacbio read overlapping)
minimap2 -x ava-ont -t8 WT_minion_allfiles.fasta WT_minion_allfiles.fasta | gzip -1 > WT_minion_fastq_allfiles.paf.gz

# Concatenate pieces of read sequences to generate the final sequences.
# Thus the per-base error rate is similar to the raw input reads. Make sure you correct your reads.
# Layout
miniasm -f WT_minion_allfiles.fasta WT_minion_fastq_allfiles.paf.gz > reads.gfa

# Convert gfa file to fasta file.
awk '/^S/{print ">"$2"\n"$3}' reads.gfa | fold > assembly/miniasm/F.venenatum/WT_minion/WT_minion.fa
```

Quast, busco and kat were run to assess the assembly quality (optionally at this stage, racon will correct errors after)

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/WT_minion.fa); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

```bash
mv assembly/miniasm/F.venenatum/WT_minion/WT_minion.fa assembly/miniasm/F.venenatum/WT_minion/WT_minion_miniasm.fa
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/WT_minion_miniasm.fa); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

Kat tool was used to assess level of error and duplications in the genome assemblies generated. Mapleson et al., 2016.

```bash
for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/WT_minion.fa); do
  Strain=$(echo $Assembly| rev | cut -d '/' -f2 | rev )
  Organism=$(echo $Assembly | rev | cut -d '/' -f3 | rev)
  echo "$Organism - $Strain"
  IlluminaDir=$(ls -d qc_dna/paired/*/Hg199)
  cat $IlluminaDir/F/*_trim.fq.gz > $IlluminaDir/F/F_trim_appended.fq.gz
  cat $IlluminaDir/R/*_trim.fq.gz > $IlluminaDir/R/R_trim_appended.fq.gz
  ReadsF=$(ls $IlluminaDir/F/F_trim_appended.fq.gz)
  ReadsR=$(ls $IlluminaDir/R/R_trim_appended.fq.gz)
  OutDir=$(dirname $Assembly)/kat
  Prefix="${Strain}"
  ProgDir=/home/gomeza/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/kat
  qsub $ProgDir/sub_kat.sh $Assembly $ReadsF $ReadsR $OutDir $Prefix 200
done
```

after KAT jobs have finished running, then remove appended trimmed reads
```bash
rm qc_dna/paired/*/*/*/F_trim_appended.fq.gz
rm qc_dna/paired/*/*/*/R_trim_appended.fq.gz
```

### Error correction using racon:

```bash
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/WT_minion.fa); do
    Strain=WT
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ReadsFq=$(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz)
    Iterations=10
    OutDir=$(dirname $Assembly)"/racon_$Iterations"
    ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/racon
    sbatch $ProgDir/sub_racon.sh $Assembly $ReadsFq $Iterations $OutDir
  done
```

Quast and busco were run to assess the effects of racon on assembly quality:


```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/WT_minion_racon_round_10.fasta); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

```bash
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
touch tmp.txt
for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/WT_minion_racon_round_10.fasta); do
  OutDir=$(dirname $Assembly)
  $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_miniasm_racon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/WT_miniasm_racon10_renamed.fasta); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

### Assemblies were polished using Pilon
Although three libraries were available, the first contained a relatively small amount of data and was not used for correction.


```bash
for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/WT_miniasm_racon10_renamed.fasta); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev | sed 's/_minion//g')
IlluminaDir=$(ls -d ../oldhome/groups/harrisonlab/project_files/fusarium_venenatum/qc_dna/paired/$Organism/$Strain)
echo $Strain
echo $Organism
TrimF2_Read=$(ls $IlluminaDir/F/*_trim.fq.gz | head -n2 | tail -n1);
TrimR2_Read=$(ls $IlluminaDir/R/*_trim.fq.gz | head -n2 | tail -n1);
TrimF3_Read=$(ls $IlluminaDir/F/*_trim.fq.gz | head -n3 | tail -n1);
TrimR3_Read=$(ls $IlluminaDir/R/*_trim.fq.gz | head -n3 | tail -n1);
echo $TrimF2_Read
echo $TrimR2_Read
echo $TrimF3_Read
echo $TrimR3_Read
OutDir=$(dirname $Assembly)
Iterations=10
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/pilon
sbatch $ProgDir/sub_pilon_2_libs.sh $Assembly $TrimF2_Read $TrimR2_Read $TrimF3_Read $TrimR3_Read $OutDir $Iterations
done
```

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/pilon_5.fasta); do
    OutDir=$(dirname $Assembly)/pilon5
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```


```bash
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/pilon_5.fasta); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

```bash
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
touch tmp.txt
for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/pilon_10.fasta); do
  OutDir=$(dirname $Assembly)
  $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_miniasm_pilon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/pilon/WT_miniasm_pilon10_renamed.fasta); do
    OutDir=$(dirname $Assembly)/pilon6
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```
```bash
  for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/pilon/WT_miniasm_pilon10_renamed.fasta); do
    #Strain=WT_minion
    #Organism=F.venenatum
    #echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

## SMARTdenovo

```bash
  for TrimReads in $(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz); do
    Organism=$(echo $TrimReads | rev | cut -f3 -d '/' | rev)
    Strain=WT
    Prefix="$Strain"_smartdenovo
    OutDir=assembly/SMARTdenovo/F.venenatum/"$Strain"
    mkdir -p $OutDir
    ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/SMARTdenovo
    sbatch $ProgDir/sub_SMARTdenovo.sh $TrimReads $Prefix $OutDir
  done
```

Quast, busco and kat were run to assess the assembly quality.

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT/WT_smartdenovo.dmo.lay.utg); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

### Error correction with racon

```bash
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT/WT_smartdenovo.dmo.lay.utg); do
    ReadsFq=$(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz)
    Iterations=10
    OutDir=$(dirname $Assembly)"/racon_$Iterations"
    ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/racon
    sbatch $ProgDir/sub_racon.sh $Assembly $ReadsFq $Iterations $OutDir
  done
```

```bash
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
touch tmp.txt
for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/WT_smartdenovo_racon_round_10.fasta); do
  OutDir=$(dirname $Assembly)
  $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_minion_racon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/WT_minion_racon10_renamed.fasta); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

```bash
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/WT_minion_racon10_renamed.fasta); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

### Assemblies were polished using Pilon

```bash
for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/WT_minion_racon10_renamed.fasta); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev | sed 's/_minion//g')
IlluminaDir=$(ls -d ../oldhome/groups/harrisonlab/project_files/fusarium_venenatum/qc_dna/paired/$Organism/$Strain)
echo $Strain
echo $Organism
TrimF2_Read=$(ls $IlluminaDir/F/*_trim.fq.gz | head -n2 | tail -n1);
TrimR2_Read=$(ls $IlluminaDir/R/*_trim.fq.gz | head -n2 | tail -n1);
TrimF3_Read=$(ls $IlluminaDir/F/*_trim.fq.gz | head -n3 | tail -n1);
TrimR3_Read=$(ls $IlluminaDir/R/*_trim.fq.gz | head -n3 | tail -n1);
echo $TrimF2_Read
echo $TrimR2_Read
echo $TrimF3_Read
echo $TrimR3_Read
OutDir=$(dirname $Assembly)
Iterations=10
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/pilon
sbatch $ProgDir/sub_pilon_2_libs.sh $Assembly $TrimF2_Read $TrimR2_Read $TrimF3_Read $TrimR3_Read $OutDir $Iterations
done
```

```bash
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
touch tmp.txt
for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/pilon/pilon_10.fasta); do
  OutDir=$(dirname $Assembly)
  $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_SMART_pilon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/pilon/WT_SMART_pilon10_renamed.fasta); do
    OutDir=$(dirname $Assembly)/pilon6
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```
```bash
  for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/pilon/WT_SMART_pilon10_renamed.fasta); do
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```

## Canu

```bash
	for Reads in $(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz); do
  	GenomeSz="37m"
  	Strain=WT
  	Organism=$(echo $Reads | rev | cut -f3 -d '/' | rev)
  	Prefix="$Strain"_canu
  	OutDir=assembly/canu/$Organism/"$Strain"
  	ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/canu
  	sbatch $ProgDir/submit_canu_minion.sh $Reads $GenomeSz $Prefix $OutDir
  done
```

Quast and busco were run to assess the effects of racon on assembly quality:

```bash
# Python 2.7 is needed to install Quast
  ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
  for Assembly in $(ls assembly/canu/F.venenatum/WT/WT_canu.contigs.fasta); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```

```bash
  for Assembly in $(ls assembly/canu/F.venenatum/WT_minion/WT_canu.contigs.fasta); do
    Strain=WT
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
    BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
    sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
  done
```


## Flye

```bash
  for TrimReads in $(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz); do
    Organism=$(echo $TrimReads | rev | cut -f3 -d '/' | rev)
    Strain=WT_minion
    Prefix="$Strain"_flye
    OutDir=assembly/flye/F.venenatum/"$Strain"
    mkdir -p $OutDir
    Size=
    ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/SMARTdenovo
    sbatch $ProgDir/flye.sh $TrimReads $Prefix $OutDir $Size
  done
```

```bash
# Python 2.7 is needed to install Quast
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc
for Assembly in $(ls flye/assembly.fasta); do
OutDir=$(dirname $Assembly)
sbatch $ProgDir/sub_quast.sh $Assembly $OutDir
done
```

```bash
for Assembly in $(ls flye/assembly.fasta); do
Strain=WT_minion
Organism=F.venenatum
echo "$Organism - $Strain"
ProgDir=/home/gomeza/git_repos/tools/gene_prediction/busco
BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
sbatch $ProgDir/sub_busco.sh $Assembly $BuscoDB $OutDir
done
```

```bash
ProgDir=/home/gomeza/git_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
touch tmp.txt
for Assembly in $(ls assembly/flye/F.venenatum/WT_minion/racon_10/racon_round_10.fasta); do
OutDir=$(dirname $Assembly)
$ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_flye_racon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
for Assembly in $(ls assembly/flye/F.venenatum/WT_minion/racon_10/WT_flye_racon10_renamed.fasta); do
Strain=WT_minion
Organism=F.venenatum
echo "$Organism - $Strain"
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Gene_prediction
BuscoDB=$(ls -d /projects/oldhome/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd9
sbatch $ProgDir/busco.sh $Assembly $BuscoDB $OutDir
done
```

```bash
# Python 2.7 is needed to install Quast
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
for Assembly in $(ls assembly/flye/F.venenatum/WT_minion/racon_10/WT_flye_racon10_renamed.fasta); do
OutDir=$(dirname $Assembly)
sbatch $ProgDir/quast.sh $Assembly $OutDir
done
```

```bash
for Assembly in $(ls assembly/flye/F.venenatum/WT_minion/assembly.fasta); do
Strain=WT_minion
Organism=F.venenatum
echo "$Organism - $Strain"
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Gene_prediction
BuscoDB=$(ls -d /projects/dbBusco/sordariomycetes_odb10)
OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd10
sbatch $ProgDir/busco.sh $Assembly $BuscoDB $OutDir
done
```


## Sequencing correction with Medaka

```
Nanopolish and Medaka were used for sequencing error correction. Medaka outperformed Nanopolish in both speed and correction for this dataset.
Nanopolish generated large number of unnecessary files that were deleted, along with scripts used to make things clearer.
If you are interested to run nanopolish the bioinformatics_tools git repo contains instructions
Note - Assembly process would be: assembly, racon, medaka and pilon
```

```bash
conda activate medaka
  for Assembly in $(ls assembly/canu/F.venenatum/WT_minion/*_canu.contigs.fasta); do
    ReadsFq=$(ls qc_dna/minion/F.venenatum/WT/WT_minion_allfiles.fastq.gz)
    OutDir=$(dirname $Assembly)/medaka
    ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Genome_assemblers
    sbatch $ProgDir/medaka.sh $Assembly $ReadsFq $OutDir
  done


for Assembly in $(ls assembly/canu/F.venenatum/WT_minion/medaka/medaka/consensus.fasta); do
  IlluminaDir=$(ls -d ../oldhome/groups/harrisonlab/project_files/fusarium_venenatum/qc_dna/paired/F.venenatum/WT)
  TrimF1_Read=$(ls $IlluminaDir/F/*S1_L001_R1_001_trim.fq.gz | head -n1)
  TrimR1_Read=$(ls $IlluminaDir/R/*S1_L001_R2_001_trim.fq.gz | head -n1)
  TrimF2_Read=$(ls $IlluminaDir/F/*S2_L001_R1_001_trim.fq.gz | head -n1)
  TrimR2_Read=$(ls $IlluminaDir/R/*S2_L001_R2_001_trim.fq.gz | head -n1)
  TrimF3_Read=$(ls $IlluminaDir/F/*S3_L001_R1_001_trim.fq.gz | head -n1)
  TrimR3_Read=$(ls $IlluminaDir/R/*S3_L001_R2_001_trim.fq.gz | head -n1)
  OutDir=$(dirname $Assembly)/pilon
  Iterations=10
  ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Genome_assemblers/pilon
  sbatch -p himem $ProgDir/pilon_3_libs.sh $Assembly $TrimF1_Read $TrimR1_Read $TrimF2_Read $TrimR2_Read $TrimF3_Read $TrimR3_Read $OutDir $Iterations
done
```

```bash
for Assembly in $(ls assembly/*/F.venenatum/WT_minion/racon_10/medaka/medaka/consensus.fasta); do
    Organism=$(echo $Assembly | rev | cut -f6 -d '/' | rev)
    Strain=$(echo $Assembly | rev | cut -f5 -d '/' | rev | sed 's/_minion//g')
    IlluminaDir=$(ls -d ../oldhome/groups/harrisonlab/project_files/fusarium_venenatum/qc_dna/paired/$Organism/$Strain)
    echo $Strain
    echo $Organism
    TrimF2_Read=$(ls $IlluminaDir/F/FvenWT_S2_L001_R1_001_trim.fq.gz | head -n2 | tail -n1);
    TrimR2_Read=$(ls $IlluminaDir/R/FvenWT_S2_L001_R2_001_trim.fq.gz | head -n2 | tail -n1);
    TrimF3_Read=$(ls $IlluminaDir/F/FvenWT_S3_L001_R1_001_trim.fq.gz | head -n3 | tail -n1);
    TrimR3_Read=$(ls $IlluminaDir/R/FvenWT_S3_L001_R2_001_trim.fq.gz | head -n3 | tail -n1);
    echo $TrimF2_Read
    echo $TrimR2_Read
    echo $TrimF3_Read
    echo $TrimR3_Read
    OutDir=$(dirname $Assembly)/pilon
    Iterations=10
    ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Genome_assemblers/pilon
    sbatch $ProgDir/sub_pilon_2_libs.sh $Assembly $TrimF2_Read $TrimR2_Read $TrimF3_Read $TrimR3_Read $OutDir $Iterations
done

for Assembly in $(ls assembly/*/F.venenatum/WT_minion/racon_10/medaka/medaka/consensus.fasta); do
    Strain=WT_minion
    Organism=F.venenatum
    echo "$Organism - $Strain"
    ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Gene_prediction
    BuscoDB=$(ls -d /projects/dbBusco/sordariomycetes_odb10)
    OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd10
    sbatch $ProgDir/busco.sh $Assembly $BuscoDB $OutDir
done

ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
for Assembly in $(ls assembly/*/F.venenatum/WT_minion/racon_10/medaka/medaka/consensus.fasta); do
    OutDir=$(dirname $Assembly)
    sbatch $ProgDir/quast.sh $Assembly $OutDir
done
```

```bash
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
touch tmp.txt
for Assembly in $(ls assembly/SMARTdenovo/F.venenatum/WT_minion/racon_10/medaka/medaka/pilon/pilon_10.fasta); do
    OutDir=$(dirname $Assembly)
    $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_SMARTdenovo_medaka_pilon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt

ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
touch tmp.txt
for Assembly in $(ls assembly/miniasm/F.venenatum/WT_minion/racon_10/medaka/medaka/pilon/pilon_10.fasta); do
OutDir=$(dirname $Assembly)
$ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_miniasm_medaka_pilon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt

ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
touch tmp.txt
for Assembly in $(ls assembly/flye/F.venenatum/WT_minion/racon_10/medaka/medaka/pilon/pilon_10.fasta); do
OutDir=$(dirname $Assembly)
$ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/WT_flye_medaka_pilon10_renamed.fasta --coord_file tmp.txt > $OutDir/log.txt
done
rm tmp.txt
```

```bash
for Assembly in $(ls assembly/*/F.venenatum/WT_minion/racon_10/medaka/medaka/pilon/*_medaka_pilon10_renamed.fasta); do
Strain=WT_minion
Organism=F.venenatum
echo "$Organism - $Strain"
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Gene_prediction
BuscoDB=$(ls -d /projects/dbBusco/sordariomycetes_odb10)
OutDir=$(dirname $Assembly)/busco_sordariomycetes_obd10
sbatch $ProgDir/busco.sh $Assembly $BuscoDB $OutDir
done
```

```bash
# Python 2.7 is needed to install Quast
ProgDir=/home/gomeza/git_repos/scripts/bioinformatics_tools/Assembly_qc
for Assembly in $(ls assembly/*/F.venenatum/WT_minion/racon_10/medaka/medaka/pilon/*_medaka_pilon10_renamed.fasta); do
OutDir=$(dirname $Assembly)
sbatch $ProgDir/quast.sh $Assembly $OutDir
done
```